packages/core/src/shared/crashMonitoring.ts

```diff
@@ -61,22 +61,33 @@ const className = 'CrashMonitoring'
  *   - To mitigate this we do not run crash reporting on machines that we detect have a flaky filesystem
  */
 export class CrashMonitoring {
-    protected heartbeat: Heartbeat | undefined
-    protected crashChecker: CrashChecker | undefined
+    protected heartbeat: Heartbeat
+    protected crashChecker: CrashChecker
 
     constructor(
         private readonly state: FileSystemState,
         private readonly checkInterval: number,
         private readonly isDevMode: boolean,
         private readonly isAutomation: boolean,
-        private readonly devLogger: Logger | undefined,
-        private readonly timeLag: TimeLag
-    ) {}
+        private readonly devLogger: Logger | undefined
+    ) {
+        this.heartbeat = new Heartbeat(this.state, this.checkInterval, this.isDevMode)
+        this.heartbeat.onFailure(() => this.cleanup())
+
+        this.crashChecker = new CrashChecker(this.state, this.checkInterval, this.isDevMode, this.devLogger)
+        this.crashChecker.onFailure(() => this.cleanup())
+    }
 
     static #didTryCreate = false
     static #instance: CrashMonitoring | undefined
     /** Returns an instance of this class or undefined if any initial validation fails. */
     public static async instance(): Promise<CrashMonitoring | undefined> {
+        // - We do not support web due to some incompatibilities
+        // - We do not support cloud desktop due to false crash reports due to sleep+vpn+ssh timeouts with the IDE
+        if (isWeb() || (await getComputeEnvType()) === 'cloudDesktop-amzn') {
+            return undefined
+        }
```

Comment:
Why not local only? Wouldn't any remote environment potentially have this issue?

Reply:
There is **a lot** of noise caused specifically by cloud desktop, compared to the other ssh'd instances it is not much. This is probably since we have lots of Amazon devs using it.

The goal is to get rid of the noise while still collecting data on the others so that if we need it in the future it is there.

Reply:
would that apply to many/most "remote" situations?